bee-network/src/lib.rs

```diff
@@ -0,0 +1,24 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+//! Utilities to establish and maintain network connections with peers.
+
+#![deny(missing_docs)]
```

Comment:
```suggestion
#![deny(warnings, missing_docs)]
```

Reply:
Why is that a good idea exactly? This way builds will possibly fail with a new compiler version (where things get slowly deprecated and become warnings first)

Reply:
I am fairly sure you where the one proposing that in the first place Alex, a couple of months ago 😄 
Our CI already denies warnings anyway so it's about having a local consistency.

Reply:
I was once in favor of doing this, true, but then I learned about why that's a bad idea in general and the argument is pretty sound. Is that only a few months ago?? I feel like I am against it much longer now, but I might be wrong :smile: 

I don't understand what the CI has anything to do with it. If it would only be us and our CI compiling the code I wouldn't mind having the flag set in the code at all. I am more worried about people, that just updated to a new compiler version, cannot compile their version of the repo anymore, unless they either downgrade the compiler again, fix the warning themselves or remove the `deny` attribute, or wait until we fixed the warning for the new compiler, so that they can pull the fix.